Leke/feature/address entity

LekeMarkaj opened the pull request:
Implemented address at user and vendor. Fully functional.
Removed password at edit profile, added address, need a new page to handle the change of password.
Implemented refresh token in the backend. Leaving frontend as an option if someone could lend a hand with implementing refresh token.
If not duhet me furr veten ;(

DrenStatovci commented:
### Pull Request Review
Overall mir jon tu bo krejt ska naj error.
Hallall!!!

Mirepo disa sene qe i kisha adresu jane (jo domosdoeshmerisht me u permirsu krejt veq pi permeni):

1.  Formati i Adresave. Opsioni qe me plotsu edhe state edhe country, kjo isha apliku mashum ne USA jo qaq shum ne shtete tjera. Kishim mujt me bo veq optional state a qa po di (nuk o qaq e rendsishme niher veq mu dok qe ja vlen me permen).
2. Update i adreses nuk po me funksionon tek vendors.
3. Te user signup nuk jon fushat per adrese, vetem te edit user jane

Qito sene me mran ne sy tjerat mu doken mire.

Faleminderit per punen e palodhshme

Liburn-Krasniqi commented:
### To add to what @DrenStatovci  said earlier:
### 1. The `State` attribute should be optional, and it **is**. (only for user though?)

Here's a screenshot from prisma studio of the `State` attribute (**User** Address):
<img width="190" alt="image" src="https://github.com/user-attachments/assets/89108f6a-7c3e-42b4-acf9-0e45314b78b6" />



Here is me trying to create a Vendor without the `State` attribute for the Vendor Address:
<img width="396" alt="image" src="https://github.com/user-attachments/assets/e0db5fad-6046-43d0-b3c0-1f3f0b36f2ec" />

Looks like it is optional on the database level, but there is some validation stopping us from not inputing the `State`
Here is a screenshot from prisma studio of the `State` attribute (**Vendor** Address):
<img width="189" alt="image" src="https://github.com/user-attachments/assets/f493f866-c5d0-493d-9ebe-9e7a09c2c2dd" />

**💡**  A good idea would be to have a visual indicator on the UI that tells us that is is optional, e.g. a placeholder that just says 
"(Optional)"

### 2. Updating vendor address doesnt work:
(Trying to change `Country` from Kosovo to Hungary)
<img width="820" alt="image" src="https://github.com/user-attachments/assets/4cff0c30-b459-4d2a-a048-c65735611398" />
<img width="1172" alt="image" src="https://github.com/user-attachments/assets/b5bee1e5-5b9c-4e56-ac36-322fbdf8111f" />

_Also notice how changing the `State` attribute is not an option_

### 3. Yes there is no way to assign an address to a user just signing up.


### Also i ran into this problem:
<img width="554" alt="image" src="https://github.com/user-attachments/assets/ba1c4d3d-cced-44df-8de7-629bc5a907b0" />

Don't know if it was there before your feature, will check later. (Probably have to delete address with user and JWT tokens)

LekeMarkaj commented:
Updated requests:
1. I let the user to fill the data in the profile. so its not to much fields in the register.
2. Fixed the bug of update vendor.
3. I bona krejt fields tadreses optional.
4. fixed delete bug

LekeMarkaj commented:
Fixed bugs.
Implemented refresh tokens. How to work with them and how to secure Paths, please check EditProfile component and anything related to it.
THANKS FOR THE FEEDBACK, hala munet me pas bugs just fyi :)

Liburn-Krasniqi commented:
### Address doesn't show on vendor:
<img width="1318" alt="image" src="https://github.com/user-attachments/assets/f8b4a374-14f1-403c-8ad4-568c35009959" />

### It exists on the backend though:
<img width="1200" alt="image" src="https://github.com/user-attachments/assets/5b294406-e1be-4116-a8bc-db4e997b6e06" />

##

Otherwise its good thanks for the work!

Letting this sit here for a little until other colleagues get the chance to see it.

Liburn-Krasniqi commented:
### Recall on last comment
Last comment turned out to be an issue with me (forgot to run ` npx prisma migrate dev`), the code worked correctly. Sorry for halting the progress :(
